src/coreComponents/physicsSolvers/multiphysics/PoroelasticSolverEmbeddedFractures.cpp

```diff
@@ -0,0 +1,834 @@
+
+          // transformation determinant
+          arrayView2d< real64 const > const & detJ = elementSubRegion->detJ();
+
+          // Fill in equilibrium operator
+          arrayView1d< real64 const > const & cellVolume = elementSubRegion->getElementVolume();
+          real64 hInv = fractureSurfaceArea[k] / cellVolume[cellElementIndex];  // AreaFrac / cellVolume
+          m_fracturesSolver->AssembleEquilibriumOperator( eqMatrix, embeddedSurfaceSubRegion, k, hInv );
+
+          // Dof number of jump enrichment
+          for( int i= 0; i < embeddedSurfaceSubRegion.numOfJumpEnrichments(); i++ )
+          {
+            jumpEqnRowIndices[i] = jumpDofNumber[k] + i - rankOffset;
+            jumpColIndices[i]    = jumpDofNumber[k] + i;
+            w( i ) = w_global[k][i] + 0*dw_global[k][i];
+          }
+
+          arrayView1d< globalIndex const > const matrixPresDofNumber =
+            elementSubRegion->getReference< array1d< globalIndex > >( pressureDofKey );
+
+          globalIndex matrixPressureColIndex = matrixPresDofNumber[cellElementIndex];
+
+          arrayView1d< real64 const > const & matrixPres =
+            elementSubRegion->getReference< array1d< real64 > >( FlowSolverBase::viewKeyStruct::pressureString );
+          pm[0] = matrixPres[cellElementIndex];
+
+          string const & solidName = m_solidSolver->solidMaterialNames()[embeddedSurfacesToCells.m_toElementRegion[k][0]];
+          SolidBase const & solid = GetConstitutiveModel< SolidBase >( *elementSubRegion, solidName );
+
+          real64 const biotCoefficient = solid.getReference< real64 >( "BiotCoefficient" );
+
+          // 1. Assembly of element matrices
+          Kwpm_elem.setValues< serialPolicy >( 0 );
+          Kwpm_gauss.setValues< serialPolicy >( 0 );
+          BlasLapackLA::matrixTMatrixMultiply( identityVec, eqMatrix, Kwpm_gauss );
+
+          // Compute traction
+          ContactRelationBase const * const
+          contactRelation = constitutiveManager->GetGroup< ContactRelationBase >( m_fracturesSolver->getContactRelationName() );
+
+          // check if fracture element is open
+          bool open = w[0] >= 0 ? true : false;
+
+          contactRelation->computeTraction( fracturePres[k], w, fractureSurfaceArea[k], tractionVec, open );
+
+          contactRelation->dTraction_dPressure( fractureSurfaceArea[k], dTdpf, open );
+
+          for( integer q=0; q<fe.getNumQuadraturePoints(); ++q )
+          {
+            const realT detJq = detJ[cellElementIndex][q];
+
+            // No neg coz the effective stress is total stress - porePressure
+            // and all signs are flipped here.
+            BlasLapackLA::matrixScale( biotCoefficient * detJq, Kwpm_gauss );
+            BlasLapackLA::matrixMatrixAdd( Kwpm_gauss, Kwpm_elem, 1 );
+          }
+
+          BlasLapackLA::matrixVectorMultiply( Kwpm_elem, pm, Rfrac, 1, 1 );
+          BlasLapackLA::vectorVectorAdd( tractionVec, Rfrac, 1 );
```

Comment:
Do you need to use `BlackLapackLA` functions, or could you switch over to `LvArray::tensorOps`?

Reply:
So, for now the equilibrium operator is still an array2d so I have used `BlasLapackLA`. As soon as #1147 is merged I will be able to switch to `LvArray::tensroOps` and change all these local variables to c-arrays.